ga4gh/server/datamodel/variants.py

```diff
@@ -37,6 +38,34 @@ def isUnspecified(str):
     return str == "" or str is None
 
 
+def deepGetAttr(obj, path):
+    """
+    Utility function which is similiar to std lib's getattr, but works
+    to resolve a dot-delimited path on an object. If path is not found
+    an `AttributeError` will be raised.
+    """
+    return reduce(getattr, path.split('.'), obj)
+
+
+def deepSetAttr(obj, path, val):
```

Comment:
Sorry, failed to communicate myself properly how you've approached it is fine. Sending in dot notation as a string scares me a little bit :).

I was thinking something [like this](https://gist.github.com/david4096/cdf8717768ee6d33d6424347759949e8), but don't let it hold up this PR.